File: backend/app/services/dashboard_service.py
from __future__ import annotations

from sqlalchemy.orm import Session

from app.models.ai_model import AIModel
from app.models.document import Document
from app.models.job import Job
from app.models.provider import Provider
from app.models.video import Video


class DashboardService:
    def __init__(self, db: Session) -> None:
        self.db = db

    def get_summary(self, user_id: str) -> dict:
        providers = self.db.query(Provider).filter(Provider.user_id == user_id).count()
        models = (
            self.db.query(AIModel)
            .join(Provider, Provider.id == AIModel.provider_id)
            .filter(Provider.user_id == user_id)
            .count()
        )
        documents = self.db.query(Document).filter(Document.user_id == user_id).count()
        videos = self.db.query(Video).filter(Video.user_id == user_id).count()
        jobs = self.db.query(Job).filter(Job.user_id == user_id).count()
        return {
            "providers": providers,
            "models": models,
            "documents": documents,
            "videos": videos,
            "jobs": jobs,
            "system_health": "healthy",
        }

File: frontend/core/dashboard/dashboardService.ts
import { eq } from "drizzle-orm";

import type { ChikaimaDatabase } from "../db/client.js";
import { aiModels, documents, jobs, providers, videos } from "../db/schema.js";

export interface DashboardSummary {
  providers: number;
  models: number;
  documents: number;
  videos: number;
  jobs: number;
  system_health: string;
}

export class DashboardService {
  constructor(private readonly db: ChikaimaDatabase) {}

  getSummary(userId: string): DashboardSummary {
    const providerCount = this.db.select().from(providers).where(eq(providers.userId, userId)).all().length;
    const modelCount = this.db
      .select({ id: aiModels.id })
      .from(aiModels)
      .innerJoin(providers, eq(providers.id, aiModels.providerId))
      .where(eq(providers.userId, userId))
      .all().length;
    const documentCount = this.db.select().from(documents).where(eq(documents.userId, userId)).all().length;
    const videoCount = this.db.select().from(videos).where(eq(videos.userId, userId)).all().length;
    const jobCount = this.db.select().from(jobs).where(eq(jobs.userId, userId)).all().length;

    return {
      providers: providerCount,
      models: modelCount,
      documents: documentCount,
      videos: videoCount,
      jobs: jobCount,
      system_health: "healthy",
    };
  }
}
